result/array-bracket-spacing.js

```diff
@@ -0,0 +1,118 @@
+/*eslint-env es6*/
+
+var arr = [ 'foo', 'bar' ];
+var [ x, y ] = z;
+
+var arr = ['foo', 'bar'];
+var [x,y] = z;
+/*eslint array-bracket-spacing: ["error", "never"]*/
+/*eslint-env es6*/
+
+var arr = [ 'foo', 'bar' ];
+var arr = ['foo', 'bar' ];
+var arr = [ ['foo'], 'bar'];
+var arr = [[ 'foo' ], 'bar'];
+var arr = [ 'foo',
+  'bar'
+];
+var [ x, y ] = z;
+var [ x,y ] = z;
+var [ x, ...y ] = z;
+var [ ,,x, ] = z;
+/*eslint array-bracket-spacing: ["error", "never"]*/
+/*eslint-env es6*/
+
+var arr = [];
+var arr = ['foo', 'bar', 'baz'];
+var arr = [['foo'], 'bar', 'baz'];
+var arr = [
+  'foo',
+  'bar',
+  'baz'
+];
+var arr = ['foo',
+  'bar'
+];
+var arr = [
+  'foo',
+  'bar'];
+
+var [x, y] = z;
+var [x,y] = z;
+var [x, ...y] = z;
+var [,,x,] = z;
+/*eslint array-bracket-spacing: ["error", "always"]*/
+/*eslint-env es6*/
+
+var arr = ['foo', 'bar'];
+var arr = ['foo', 'bar' ];
+var arr = [ ['foo'], 'bar' ];
+var arr = ['foo',
+  'bar'
+];
+var arr = [
+  'foo',
+  'bar'];
+
+var [x, y] = z;
+var [x,y] = z;
+var [x, ...y] = z;
+var [,,x,] = z;
+/*eslint array-bracket-spacing: ["error", "always"]*/
+/*eslint-env es6*/
+
+var arr = [];
+var arr = [ 'foo', 'bar', 'baz' ];
+var arr = [ [ 'foo' ], 'bar', 'baz' ];
+var arr = [ 'foo',
+  'bar'
+];
+var arr = [
+  'foo',
+  'bar' ];
+var arr = [
+  'foo',
+  'bar',
+  'baz'
+];
+
+var [ x, y ] = z;
```

Comment:
__[development][eslint]__
* Error - 'x' is already defined (no-redeclare)
* Error - 'y' is already defined (no-redeclare)
* Error - 'z' is not defined. (no-undef)